data/src/main/java/com/ionutconstantinnicolae/data/NewsRepository.java

```diff
@@ -0,0 +1,23 @@
+package com.ionutconstantinnicolae.data;
+
+import com.ionutconstantinnicolae.data.feature.news.local.ArticleEntity;
+import com.ionutconstantinnicolae.data.feature.news.model.Article;
+
+import java.util.List;
+
+import io.reactivex.Completable;
+import io.reactivex.Flowable;
+import io.reactivex.Single;
+import io.reactivex.annotations.NonNull;
+
+public interface NewsRepository {
+
+    @NonNull
+    Single<List<Article>> getNewsArticles();
+
+    @androidx.annotation.NonNull
+    Single<List<Article>> getArticleList();
+
+    @androidx.annotation.NonNull
+    Completable saveArticleList(List<Article> articleList);
```

Comment:
this does not seem right, saveArticleList should not be exposed for our use-case